source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/MediaRenderer.java

```diff
@@ -206,18 +239,109 @@ private ImageView renderPlayButton(
         return playButtonView;
     }
 
-    private FullscreenVideoView renderMediaPlayer(
-            Context context,
-            ViewGroup viewGroup,
-            Media media,
-            HostConfig hostConfig)
+    static class ControllerVisibilityListener implements StyledPlayerView.ControllerVisibilityListener
+    {
+        @Override
+        public void onVisibilityChanged(int visibility) {
+
+        }
+    }
+
+    static class PlayerListener implements Player.Listener
+    {
+        private MediaOnCompletionListener m_onCompletionListener = null;
+
+        public PlayerListener(MediaOnCompletionListener onCompletionListener)
+        {
+            m_onCompletionListener = onCompletionListener;
+        }
+
+        @Override
+        public void onEvents(Player player, Player.Events events)
+        {
+            if (events.contains(EVENT_PLAYBACK_STATE_CHANGED))
+            {
+                m_onCompletionListener.onCompletion(null);
+            }
+        }
+    }
+
+    private View renderNewMediaPlayer(
+        RenderedAdaptiveCard renderedCard,
+        Context context,
+        ViewGroup viewGroup,
+        Media media,
+        ICardActionHandler cardActionHandler,
+        HostConfig hostConfig)
+    {
+        StyledPlayerView playerView = new StyledPlayerView(context);
+        playerView.setUseController(true);
+        playerView.setControllerHideOnTouch(true);
+        playerView.setResizeMode(AspectRatioFrameLayout.RESIZE_MODE_FIXED_WIDTH);
+        playerView.setControllerVisibilityListener(new ControllerVisibilityListener());
+        playerView.setShowSubtitleButton(true);
+        playerView.setControllerShowTimeoutMs(4000);
```

Comment:
where is 4000 coming from?

Reply:
random number, we don't specify requirements for the timeout for controllers to be hidden so I assume 4 seconds should be good enough

Reply:
how about the javascript? you can align with it

Reply:
Feel free to close it if you javascript doesn't define it

Reply:
couldn't find any definition for the js renderer